### PR TITLE
End-to-end testing

### PR DESCRIPTION
**Pull request by markspolakovs**
This PR adds the infrastructure for end-to-end testing the full prototype. The tests run against the front-end, which talks to the real DMAS/DroneManager APIs, running against a real PostgreSQL database.

Currently the tests are very basic (essentially "verify the front-end doesn't crash when loading the map screen"), but more can be added.

The tests are written using Playwright's Python API (https://playwright.dev/python/docs/writing-tests). If a test fails, Playwright will output a video and a trace, which can be visualised using the Trace Viewer at https://trace.playwright.dev. It'll also save the logs of all the containers, and upload both to the action artifacts.

To avoid needing to train the human classification weights each time, they're currently fetched from a Google Cloud Storage bucket on my account.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ services:
       POSTGRES_PASSWORD: "heresapassword"
     volumes:
       - "dbfiles:/var/lib/postgresql/data"
+      - "./DMAS/sql/init.sql:/docker-entrypoint-initdb.d/01-dmas.sql"
```
Comment (Laura7089) [inline docker-compose.yml:13]: Does DMAS not run this on it's own? Have you tested this works ok from a clean install?

File: requirements.txt
Comment (Laura7089) [inline requirements.txt:1]: Perhaps call this something like `requirements.test.txt` or similar to make it clear that it's only for testing?

File: .github/workflows/e2e.yml
Comment (Laura7089) [inline .github/workflows/e2e.yml:1]: I'm not familiar with GH actions so I can't really approve or deny this file

File: docker-compose.yml (hunk 1 above)
Comment (markspolakovs) [reply docker-compose.yml:13]: It does not run it on its own. I've tested it on a clean install (finding and fixing https://github.com/2023-gpig-a/DMAS/pull/26 in the process)

File: requirements.txt
Comment (markspolakovs) [reply requirements.txt:1]: Good shout, done

**Review by Laura7089 — APPROVED** [review]

**Pull request merged**